### ISSUE TITLE
 Expected type hint int[]; found array in Squiz FunctionCommentSniff

### ISSUE THREAD

**Issue by JDGrimes** [squizlabs/PHP_CodeSniffer#601]
**Example code:**

```php
	/**
	 * Does foo with IDs.
	 *
	 * @param int[] $ids List of IDs.
	 */
	function foo( array $ids ) {}
```

**Expected result:**
This should be OK.

**Actual result:**

```
 4   | ERROR | Expected type hint "int[]"; found "array" for $ids
     |       | (Squiz.Commenting.FunctionComment.IncorrectTypeHint)
 ```

---

The error comes from https://github.com/squizlabs/PHP_CodeSniffer/blob/master/CodeSniffer/Standards/Squiz/Sniffs/Commenting/FunctionCommentSniff.php#L386.

**Comment by aik099** [comment]
Because it expects `integer[]` to be used. Not short `int` form.

**Comment by aik099** [comment]
Maybe we need to make error message more intelligent.

**Comment by JDGrimes** [comment]
Yeah, I think `int` is OK in the PHPDoc spec.

**Comment by aik099** [comment]
It is, but Squiz standard specifically requires long type names.

**Comment by JDGrimes** [comment]
> It is, but Squiz standard specifically requires long type names.

I see. But I guess there is a separate error for that? I think it should take `int` (and `bool`) into account here so that it is more configurable.

Also, this actually isn't limited to short types. The same error would be given for `integer[]` (or `string[]` or `object[]` etc.).

**Comment by aik099** [comment]
Then this is surely a bug.

**Comment by NikolaySl** [comment]
The same for
```php
    /**
     * Does foo with MyClass objects.
     *
     * @param MyClass[] $list List of objects.
     */
    function foo( array $list ) {}
```
gives
```
 4 | ERROR | Expected type hint "MyClass[]"; found "array" for $list

**Comment by aik099** [comment]
That is another non-related bug with forcing user to use typehint, that will end up with syntax error in PHP. Please report it separately.

**Comment by NikolaySl** [comment]
https://github.com/squizlabs/PHP_CodeSniffer/issues/624
